File: 0280-wiggle-sort/0280-wiggle-sort.py
class Solution(object):
    def wiggleSort(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        for i in range(1, len(nums)):
            if (i & 1 == 0 and nums[i-1] < nums[i]) or (i & 1 == 1 and nums[i-1] > nums[i]):
                nums[i-1], nums[i] = nums[i], nums[i-1]
        
        return nums

File: 0280-wiggle-sort/0280-wiggle-sort.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        
        //sort(nums.begin(), nums.end());
        //int n = nums.size();
        
        for(int i = 1; i < nums.size(); i++) {
            if ( ((i % 2 == 0) && (nums[i-1] < nums[i])) ||
                 ((i % 2 == 1) && (nums[i-1] > nums[i]))
                )
            swap(nums[i-1], nums[i]);
        }   
    }
};